### PR TITLE
Update workflow to use admin token to bypass ruleset

### LINKED ISSUE #4: DOM Conversion
Problem: Input of two or more LDDs to LDDTool (an LDD stack) results in null pointer exception. 
Analysis: LDDTool tried to find LDD nameSpaceID for "Import" statements in configured PDS4 nameSpaceID array. 
Solution: Use masterAllSchemaFileSortMap instead of masterSchemaFileSortMap. The latter includes only those nameSpaces to be written to XML Schema file.

### LINKED ISSUE #1056: Automated generation of PDS Namespace Registry files from CSV update fails
### Checked for duplicates

Yes - I've already checked

### 🐛 Describe the bug

When I did merged an updated `docs/namespace-registry/pds-namespace-registry.csv` to the `main` branch, I noticed that the expected action kicked off but failed.

### 🕵️ Expected behavior

I expected the action to complete successfully.

### 📜 To Reproduce

1. Go to https://github.com/NASA-PDS/pds4-information-model/actions/workflows/generate-namespace-registry-files.yml
2. Click "Run workflow"
3. Read output

### 🖥 Environment Info

N/A

### 📚 Version of Software Used

N/A

### 🩺 Test Data / Additional context

Follow-up to https://github.com/NASA-PDS/pds4-information-model/issues/1053 and https://github.com/NASA-PDS/pds4-information-model/pull/1054.

The issue is known; see "Engineering Details" below.

### 🦄 Related requirements

🦄 #xyz

### ⚙️ Engineering Details

The repo ruleset doesn't allow for changes to be committed directly to the `main` branch and must instead be made through a PR. The question is whether to...
1. create a PR which 
   a. must be manually reviewed, approved, and merged; or
   b. is automatically merged after requirements are satisfied; or
2. bypass the ruleset (with a custom app token?)### 🎉 Integration & Test

_No response_

### PR DESCRIPTION
**Pull request by c-suh**
## 🗒️ Summary

Used https://github.com/pds-data-dictionaries/ldd-template/blob/main/.github/workflows/ldd-ci.yml as reference.

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [ ] AI generated substantial portions of this code

Estimated % of code influenced by AI: ___ %

## ♻️ Related Issues

Resolves #1056

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: .github/workflows/generate-namespace-registry-files.yml [hunk 1]
```diff
@@ -44,11 +48,11 @@ jobs:
...
         run: |
-          git config --local user.email "github-actions[bot]@users.noreply.github.com"
-          git config --local user.name "github-actions[bot]"
+          git config --local user.email "pdsen-ci@jpl.nasa.gov"
+          git config --local user.name "PDSEN CI Bot"
           git add docs/namespace-registry/pds-namespace-registry.xlsx docs/namespace-registry/pds-namespace-registry.pdf
-          git commit -m "Auto-generate namespace registry files from CSV [skip ci]"
-          git push
+          git commit -m "Auto-generated namespace registry files from CSV by PDSEN CI Bot [skip ci]"
+          git push origin HEAD
```
Comment (nutjob4life) [inline .github/workflows/generate-namespace-registry-files.yml]: Just to make things extra explicit (if the workflow is everrun from a PR context or detached `HEAD`):

    git push origin HEAD:${{github.ref_name}}

**Review by nutjob4life — APPROVED** [review]
YAML delta: ✓
Visual inspection: ✓ (note: _only_ visual insepction performed)
Zero changes needed: ✓
Approval: ✅
Note: there's one tiny comment, easily ignored. After all, this PR is already approved!

File: .github/workflows/generate-namespace-registry-files.yml [hunk 2]
```diff
@@ -12,6 +12,8 @@ on:
 jobs:
   generate-files:
     runs-on: ubuntu-latest
+    # Prevent infinite loops if bot user triggers workflow
+    if: github.actor != 'pdsen-ci'
```
[the remark this answers is not in the document]
Comment (c-suh) [reply .github/workflows/generate-namespace-registry-files.yml]: Removed this check as don't want to restrict further automation.

File: .github/workflows/generate-namespace-registry-files.yml [hunk 3]
```diff
@@ -21,7 +23,9 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v4
         with:
-          # Fetch at least the immediate parents for PR context
+          # Use admin token to bypass branch protection rules
+          token: ${{ secrets.ADMIN_GITHUB_TOKEN || github.token }}
+          # Fetch at least the immediate parents for context
           fetch-depth: 2
```
[the remark this answers is not in the document]
Comment (c-suh) [reply .github/workflows/generate-namespace-registry-files.yml:36]: True. Resolved in commit 31029d9f6c87dc91de785e8e05d79ecd7caed0a5.

File: .github/workflows/generate-namespace-registry-files.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (c-suh) [reply .github/workflows/generate-namespace-registry-files.yml]: Ignoring because want to leave it open to whichever branch it's run on. nutjob4life's suggestion below is much more preferable.

**Comment by c-suh** [comment]
@nutjob4life many, many thanks for your stellar and speedy reviews! I've incorporated your suggestion and will merge this forthwith.
